stores/metadata.go

```diff
@@ -2233,6 +2221,13 @@ func sqlConcat(db *gorm.DB, a, b string) string {
 	return fmt.Sprintf("CONCAT(%s, %s)", a, b)
 }
 
+func sqlRandomTimestamp(db *gorm.DB, now time.Time, min, max time.Duration) string {
```

Comment:
I'd change this to return a `clause.Expr` and then instead of `Sprintf` use `gorm.Expr`.